### PR TITLE
Add tests for api views

### PR DESCRIPTION
**Pull request by flowerncsu**

### CODE REVIEW

File: api/tests.py [hunk 1]
```diff
@@ -1,3 +1,158 @@
...
+
+    def setUp(self):
+        self.geometry_store = GeometryStore()
+        self.geometry_store.save()
+        point = PointGeometry(geom=Point(30, 40))
+        point.save()
+        self.geometry_store.points.add(point)
+        self.geometry_store.save()
+        self.url = reverse('api:geometrystore-detail', args=[self.geometry_store.identifier])
+        self.project = ProjectFactory()
```
Comment (mlavin) [inline api/tests.py]: It isn't clear from the tests why this setup is needed. What happens if there isn't a project?

File: api/tests.py [hunk 2]
```diff
@@ -1,3 +1,158 @@
...
+
+
+class TestGeometryStoreCentroidViewSet(TestCase):
+
+    def setUp(self):
+        self.geom_with_published_project = GeometryStore()
+        self.geom_with_published_project.save()
+        point = PointGeometry(geom=Point(20, 30))
+        point.save()
+        self.geom_with_published_project.points.add(point)
```
Comment (mlavin) [inline api/tests.py]: Similar feedback here. Why is all of these setup needed? It doesn't appear to be used in the tests.

File: api/tests.py [hunk 3]
```diff
@@ -1,3 +1,321 @@
...
+        response = self.client.get(url)
+        self.assertEqual(len(json.loads(response.content.decode())), len(ProjectStatus.STATUSES))
+
+
+class TestGeometryStoreDetailView(TestCase):
+
+    def setUp(self):
+        self.geometry_store = GeometryStore()
+        self.geometry_store.save()
+        point = PointGeometryFactory()
```
Comment (mlavin) [inline api/tests.py]: Worth noting that by default `factory_boy` calls save for these factory calls so the additional save after each one shouldn't be needed.

Comment (flowerncsu) [inline api/tests.py]: Yeah, that was my thought, too, except that when I didn't have that save(), I got errors having to do with this model being on database "None" and something else being on database "default" ....

Comment (mlavin) [inline api/tests.py]: To clarify, that's the default behavior when using the `DjangoModelFactory` which would be preferred when working with Django models. It doesn't look like that's the case for all of these.

File: newsilkroad/urls.py [hunk 4]
```diff
@@ -29,6 +29,7 @@
     url(r'^', include('website.urls')),
     url(r'^filer/', include('filer.urls')),
     url(r'^api/', include('api.urls')),
+    url(r'^infrastructure/', include('infrastructure.urls')),
```
Comment (mlavin) [inline newsilkroad/urls.py]: These are included under  `/database/` in `website/urls.py` already. We shouldn't need to add them again.

Comment (flowerncsu) [inline newsilkroad/urls.py]: Fixed

**Comment by mlavin** [comment]
Remaining issues from my perspective:

- [ ] Test factories should be using `DjangoModelFactory`, removing the additional `.save()` calls
- [ ] Coverage needs to be improved for the filters (infrastructure in particular)
- [ ] `DynamicFieldsMixin` is missing the primary logical branch of coverage

**Pull request merged**